src/components/TodoItem.tsx

```diff
@@ -0,0 +1,181 @@
+/* eslint-disable jsx-a11y/label-has-associated-control */
+/* eslint-disable jsx-a11y/control-has-associated-label */
+
+import React, { useEffect, useRef, useState } from 'react';
+import cn from 'classnames';
+import { Todo } from '../types/Todo';
+
+interface Props {
+  todosToDelete?: number[];
+  todo: Todo;
+  onTodosToDelete: React.Dispatch<React.SetStateAction<number[]>>;
+  onUpdate: (data: Todo) => Promise<void>;
+}
+
+export const TodoItem: React.FC<Props> = ({
+  todosToDelete,
+  todo,
+  onTodosToDelete,
+  onUpdate,
+}) => {
+  const [isEditing, setIsEditing] = useState(false);
+  const [newTitle, setNewTitle] = useState(todo.title);
+  const [isUpdating, setIsUpdating] = useState(false);
+  const [isChecked, setIsChecked] = useState(todo.completed);
```

Comment:
why do you need the `isChecked` state? I haven't seen that you use it

Reply:
Why do you need the `isChecked` state? You can use `todo.completed`